runtime/lua/vim/provider/health.lua

```diff
@@ -366,7 +366,8 @@ local function check_for_pyenv()
   local pyenv_root = vim.fn.resolve(os.getenv('PYENV_ROOT') or '')
 
   if pyenv_root == '' then
-    pyenv_root = vim.fn.system({ pyenv_path, 'root' })
+    local p = vim.system({ pyenv_path, 'root' }):wait()
```

Comment:
I think it's better add stderr/code error handling.

Reply:
Done, add check if  `p.code~=0` and call `health.warn` to report error

Reply:
Just checking the exit code is not enough. I prefer also add stderr check.

Reply:
Are you mean stderr maybe nil? I don't understand,  the content of stderr is irrelevant with neovim, I prefer just show it to user

Reply:
some programs will output errors to stderr but the exit code is normal. not sure if it will appear here. so I suggest checking stderr.

Reply:
No, `pyenv root` just a warapper of [`pyenv-root`](https://github.com/pyenv/pyenv/blob/master/libexec/pyenv-root), before execute `pyenv-root` , pyenv set the `PYENV_PATH` variable in script scope, so the `pyenv-root` only need call `echo "$PYENV_ROOT"`  as show in the `pyenv-root` script's content, the `echo` command isn't the case which you mention.

In the other words, `pyenv root` just a complex `echo $PYENV_ROOT`. So, I think it is enought to display the stderror

To be honest, if `pyenv` is a executable (which is ensured in previous code), `pyenv root` must work as our want,  refer to https://github.com/pyenv/pyenv/tree/master?tab=readme-ov-file#locating-pyenv-provided-python-installations